### PR TITLE
New "Geo" optional dependencies

### PR DESCRIPTION
**Pull request by odscjames**
https://github.com/OpenDataServices/flatten-tool/issues/424

### CODE REVIEW

**Comment by odscjames** [comment]
@jpmckinney @michaelwood @Bjwebb Does this framing of "Geo" work for everyone?

[ I still need to sort C.I. and coveralls so in draft ]

File: flattentool/input.py [hunk 1]
```diff
@@ -105,7 +112,11 @@ def convert_type(type_string, value, timezone=pytz.timezone("UTC"), convert_flag
         if type(value) == datetime.datetime:
             return value.date().isoformat()
         return str(value)
-    elif convert_flags.get("wkt") and type_string == "geojson":
+    elif (
+        convert_flags.get("wkt")
+        and type_string == "geojson"
+        and SHAPELY_AND_GEOJSON_LIBRARIES_AVAILABLE
```
Comment (Bjwebb) [inline flattentool/input.py:118]: Can we have a warning if the first 2 are true, but the 3rd isn't? (And similarly in `json_input.py`).

Comment (odscjames) [reply flattentool/input.py:118]: That's even worse to write tests for :-P Good point

**Comment by jpmckinney** [comment]
Works for me!

**Comment by Bjwebb** [comment]
I've added a comment about warnings, and I think generally these are lowercase (so `geo` might be better), but otherwise looking good.

**Comment by odscjames** [comment]
After a couple of minutes search via PEP8, [PEP685](https://peps.python.org/pep-0685/#specification), https://packaging.python.org/en/latest/specifications/dependency-specifiers/ and PEP345 I couldn't find a still-valid official preference explicitly stated but the old document says lower and current examples are lower case - so lower case "geo" it is.

File: flattentool/input.py (hunk 1 above)
Comment (odscjames) [reply flattentool/input.py]: Seen in tests:

```
==================================================================================================================================== warnings summary ====================================================================================================================================
flattentool/tests/test_json_input.py::test_parse_geojson
  .../flattentool/json_input.py:382: UserWarning: Install flattentool's optional geo dependencies to use geo features.
    warn(

```

**Comment by odscjames** [comment]
If  we later add lots more optional dependencies and want to test various combos, I'm not sure if running all of them in one runner or many is better. But for now, doing it in one runner makes coverage reporting much easier so we'll do that.

**Review by Bjwebb — APPROVED** [review]

**Pull request merged**